### PR TITLE
add cgropu swap usage and export as prometheus metric

### PR DESCRIPTION
**Pull request by f0**
Hi ,

this adds the swap information from memory.stat and exports this via prometheus metrics.

Question: Atm this is implemented with the memory.stat file, maybe ists better to use memory_usage_in_bytes and memsw_usage_in_bytes ?

regards f0

### CODE REVIEW

**Comment by f0** [comment]
if i take a look at the documentation https://lwn.net/Articles/432224/  i think the toalt_{rss,cache,swap} values should be used, but cadvisor uses the not total prefixed once, is there a reason?

**Comment by f0** [comment]
@vishh  any opinion about the total vs non total statistics?

**Comment by vishh** [comment]
Isn't total_ meant for hierarchical stats instead of just this cgroups' stats?

**Comment by f0** [comment]
@vishh 
 thats true, but the cgroup casn have children cgroups or not?

**Comment by vishh** [comment]
They can. Current cAdvisor API doesn't provide hierarchical stats AFAIK.

**Comment by f0** [comment]
ok, switched back to the non total value

**Comment by f0** [comment]
any chance that this can be merged ?

File: metrics/prometheus.go [hunk 1]
```diff
@@ -437,7 +444,26 @@ func NewPrometheusCollector(infoProvider infoProvider, f ContainerNameToLabelsFu
 					}
 					return values
 				},
-			}, {
+			},
+
+			{
+				name:      "container_tcp_connections_established",
+				help:      "XXX.",
```
Comment (timstclair) [inline metrics/prometheus.go]: TODO?

File: metrics/prometheus.go [hunk 2]
```diff
@@ -437,7 +444,26 @@ func NewPrometheusCollector(infoProvider infoProvider, f ContainerNameToLabelsFu
 					}
 					return values
 				},
-			}, {
+			},
+
+			{
+				name:      "container_tcp_connections_established",
```
Comment (timstclair) [inline metrics/prometheus.go]: This looks unrelated to this PR? Same below.

File: info/v1/container.go [hunk 3]
```diff
@@ -324,6 +324,10 @@ type MemoryStats struct {
 	// Units: Bytes.
 	RSS uint64 `json:"rss"`
 
+	// The amount of swap currently used by the processes in this cgroup
+	// Units: Bytes.
+	Swap uint64 `json:"swap"`
```
Comment (timstclair) [inline info/v1/container.go]: This should be added to v2 as well.

File: metrics/prometheus.go [hunk 4]
```diff
@@ -142,6 +142,13 @@ func NewPrometheusCollector(infoProvider infoProvider, f ContainerNameToLabelsFu
 					return metricValues{{value: float64(s.Memory.RSS)}}
 				},
 			}, {
+				name:      "container_memory_swap",
+				help:      "container swap usage in bytes.",
```
Comment (timstclair) [inline metrics/prometheus.go]: (nit) capitalize container

**Comment by f0** [comment]
comments fixed, unrelated stuff removed

File: info/v1/container.go (hunk 3 above)
Comment (f0) [inline info/v1/container.go]: ok i take a look

Comment (f0) [inline info/v1/container.go]: @timstclair  hm v2 uses the v1 values

**Comment by f0** [comment]
@timstclair  @vishh any outstanding issues?

**Comment by timstclair** [comment]
LGTM. Needs a rebase though.

**Comment by f0** [comment]
@timstclair  rebased
